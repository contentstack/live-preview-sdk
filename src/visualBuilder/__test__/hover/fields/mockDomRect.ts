export const mockDomRect = {
    singleHorizontal: () => ({
        bottom: 88.3984375,
        height: 54.3984375,
        left: 34,
        right: 862,
        top: 34,
        width: 828,
        x: 34,
        y: 34,
    }),
    singleLeft: () => ({
        x: 51,
        y: 51,
        width: 27.7734375,
        height: 20.3984375,
        top: 51,
        bottom: 71.3984375,
        left: 51,
        right: 78.7734375,
    }),
    singleRight: () => ({
        x: 110.7734375,
        y: 51,
        width: 53.59375,
        height: 20.3984375,
        top: 51,
        bottom: 71.3984375,
        left: 110.7734375,
        right: 164.3671875,
    }),
    singleVertical: () => ({
        bottom: 195.1953125,
        height: 90.796875,
        left: 34,
        right: 862,
        top: 104.3984375,
        width: 828,
        x: 34,
        y: 104.3984375,
    }),
    singleTop: () => ({
        x: 51,
        y: 121.3984375,
        width: 794,
        height: 20.3984375,
        top: 121.3984375,
        bottom: 141.796875,
        left: 51,
        right: 845,
    }),
    singleBottom: () => ({
        x: 51,
        y: 157.796875,
        width: 794,
        height: 20.3984375,
        top: 157.796875,
        bottom: 178.1953125,
        left: 51,
        right: 845,
    }),
    multipleLeft: () => ({
        x: 51,
        y: 228.1953125,
        width: 87.59375,
        height: 90.796875,
        top: 228.1953125,
        bottom: 318.9921875,
        left: 51,
        right: 138.59375,
    }),
    multipleRight: () => ({
        x: 170.59375,
        y: 228.1953125,
        width: 87.59375,
        height: 90.796875,
        top: 228.1953125,
        bottom: 318.9921875,
        left: 170.59375,
        right: 258.1875,
    }),
    multipleTop: () => ({
        x: 51,
        y: 368.9921875,
        width: 794,
        height: 90.796875,
        top: 368.9921875,
        bottom: 459.7890625,
        left: 51,
        right: 845,
    }),
    multipleBottom: () => ({
        x: 51,
        y: 475.7890625,
        width: 794,
        height: 90.796875,
        top: 475.7890625,
        bottom: 566.5859375,
        left: 51,
        right: 845,
    }),
    multipleChildTop: () => ({
        x: 68,
        y: 385.9921875,
        width: 760,
        height: 20.3984375,
        top: 385.9921875,
        bottom: 406.390625,
        left: 68,
        right: 828,
    }),
    multipleChildBottom: () => ({
        x: 68,
        y: 422.390625,
        width: 760,
        height: 20.3984375,
        top: 422.390625,
        bottom: 442.7890625,
        left: 68,
        right: 828,
    }),
};
