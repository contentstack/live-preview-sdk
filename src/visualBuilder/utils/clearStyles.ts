const originalStyles = new WeakMap<
    HTMLElement,
    {
        visibility?: string;
        transition?: string;
        animation?: string;
    }
>();

export function clearVisibelityStyles(element: HTMLElement) {
    const originalStyleValues = {
        visibility: element.style.visibility,
        transition: element.style.transition,
        animation: element.style.animation,
    };

    originalStyles.set(element, originalStyleValues);

    element.style.visibility = "hidden";
    element.style.transition = "none";
    element.style.animation = "none";
}

export function restoreVisibelityStyles(element: HTMLElement) {
    const storedStyles = originalStyles.get(element);

    if (storedStyles) {
        element.style.visibility = storedStyles.visibility || "";
        element.style.transition = storedStyles.transition || "";
        element.style.animation = storedStyles.animation || "";

        originalStyles.delete(element);
    }
}
