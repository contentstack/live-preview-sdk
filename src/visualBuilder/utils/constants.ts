import { FieldDataType } from "./types/index.types";

export const numericInputRegex = /^-?\d*(\.\d*)?([eE][-+]?\d*)?$/;

export const VISUAL_BUILDER_FIELD_TYPE_ATTRIBUTE_KEY = "data-cslp-field-type";

export const VISUAL_BUILDER_CHANNEL_ID = "visual-builder";

export const LIVE_PREVIEW_OUTLINE_WIDTH_IN_PX = 2;

// Minimum distance from top edge to prevent toolbar from being hidden
export const TOP_EDGE_BUFFER = 42;
export const RIGHT_EDGE_BUFFER = 180;

export const TOOLBAR_EDGE_BUFFER = 8;

export const DATA_CSLP_ATTR_SELECTOR = "data-cslp";

/**
 * The field that can be directly modified using contenteditable=true.
 * This includes all text fields like title and numbers.
 */
export const ALLOWED_INLINE_EDITABLE_FIELD: FieldDataType[] = [
    FieldDataType.SINGLELINE,
    FieldDataType.MULTILINE,
    FieldDataType.NUMBER,
];

export const ALLOWED_MODAL_EDITABLE_FIELD: FieldDataType[] = [
    FieldDataType.HTML_RTE,
    FieldDataType.MARKDOWN_RTE,
    FieldDataType.JSON_RTE,
    FieldDataType.CUSTOM_FIELD,
    FieldDataType.LINK,
    FieldDataType.ISODATE,
    FieldDataType.URL,
];

export const ALLOWED_REPLACE_FIELDS: FieldDataType[] = [
    FieldDataType.REFERENCE,
    FieldDataType.FILE,
];

export const DEFAULT_MULTIPLE_FIELDS: FieldDataType[] = [
    FieldDataType.GLOBAL_FIELD,
    FieldDataType.GROUP,
    FieldDataType.BLOCK,
];

export const mentionLimit = 20;
export const maxMessageLength = 500;
