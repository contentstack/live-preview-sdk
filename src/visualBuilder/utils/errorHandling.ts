
export function hasPostMessageError(obj: any): boolean {
    return obj?.error === true;
}