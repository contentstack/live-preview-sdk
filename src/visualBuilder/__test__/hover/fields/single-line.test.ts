import { getFieldSchemaMap } from "../../../../__test__/data/fieldSchemaMap";
import { waitForHoverOutline } from "../../../../__test__/utils";
import Config from "../../../../configManager/configManager";
import { VisualBuilder } from "../../../index";
import { FieldSchemaMap } from "../../../utils/fieldSchemaMap";
import { mockDomRect } from "./mockDomRect";
import { act, screen } from "@testing-library/preact";

vi.mock("../../../utils/visualBuilderPostMessage", async () => {
    const { getAllContentTypes } = await vi.importActual<
        typeof import("../../../../__test__/data/contentType")
    >("../../../../__test__/data/contentType");
    const contentTypes = getAllContentTypes();
    return {
        __esModule: true,
        default: {
            send: vi.fn().mockImplementation((eventName: string) => {
                if (eventName === "init")
                    return Promise.resolve({
                        contentTypes,
                    });
                return Promise.resolve();
            }),
        },
    };
});

global.ResizeObserver = vi.fn().mockImplementation(() => ({
    observe: vi.fn(),
    unobserve: vi.fn(),
    disconnect: vi.fn(),
}));

vi.mock("../../../../utils/index.ts", async () => {
    const actual = await vi.importActual("../../../../utils");
    return {
        __esModule: true,
        ...actual,
        isOpenInBuilder: vi.fn().mockReturnValue(true),
    };
});

describe("When an element is hovered in visual builder mode", () => {
    let mousemoveEvent: Event;

    beforeAll(() => {
        FieldSchemaMap.setFieldSchema(
            "all_fields",
            getFieldSchemaMap().all_fields
        );
    });

    beforeEach(() => {
        Config.reset();
        Config.set("mode", 2);
        mousemoveEvent = new Event("mousemove", {
            bubbles: true,
            cancelable: true,
        });
    });

    afterEach(() => {
        vi.clearAllMocks();
        document.getElementsByTagName("html")[0].innerHTML = "";
    });

    afterAll(() => {
        Config.reset();
    });

    describe("title field", () => {
        let titleField: HTMLParagraphElement;
        let visualBuilder: VisualBuilder;

        beforeEach(() => {
            titleField = document.createElement("p");
            titleField.setAttribute(
                "data-cslp",
                "all_fields.bltapikey.en-us.title"
            );
            titleField.getBoundingClientRect = vi
                .fn()
                .mockReturnValue(mockDomRect.singleLeft());
            document.body.appendChild(titleField);

            visualBuilder = new VisualBuilder();
        });

        afterEach(() => {
            visualBuilder.destroy();
        });

        test("should have outline and custom cursor", async () => {
            await act(() => {
                titleField.dispatchEvent(mousemoveEvent);
            });
            await waitForHoverOutline();
            expect(titleField).not.toHaveAttribute("style");
            const hoverOutline = screen.getByTestId(
                "visual-builder__hover-outline"
            );
            expect(hoverOutline).toHaveStyle(
                "top: 51px; left: 51px; width: 27.7734375px; height: 20.3984375px;"
            );

            const customCursor = document.querySelector(
                `[data-testid="visual-builder__cursor"]`
            );
            expect(customCursor).toHaveAttribute("data-icon", "singleline");
            expect(customCursor?.classList.contains("visible")).toBeTruthy();
        });
    });

    describe("single line field", () => {
        let singleLineField: HTMLParagraphElement;
        let visualBuilder: VisualBuilder;

        beforeEach(() => {
            singleLineField = document.createElement("p");
            singleLineField.setAttribute(
                "data-cslp",
                "all_fields.bltapikey.en-us.single_line"
            );
            singleLineField.getBoundingClientRect = vi
                .fn()
                .mockReturnValue(mockDomRect.singleLeft());
            document.body.appendChild(singleLineField);

            visualBuilder = new VisualBuilder();
        });

        afterEach(() => {
            visualBuilder.destroy();
        });

        test("should have outline and custom cursor", async () => {
            await act(() => {
                singleLineField.dispatchEvent(mousemoveEvent);
            });
            await waitForHoverOutline();
            expect(singleLineField).not.toHaveAttribute("style");

            const customCursor = document.querySelector(
                `[data-testid="visual-builder__cursor"]`
            );

            expect(customCursor).toHaveAttribute("data-icon", "singleline");
            expect(customCursor?.classList.contains("visible")).toBeTruthy();
        });
    });

    describe("single line field (multiple)", () => {
        let container: HTMLDivElement;
        let firstSingleLineField: HTMLParagraphElement;
        let secondSingleLineField: HTMLParagraphElement;
        let visualBuilder: VisualBuilder;

        beforeEach(() => {
            container = document.createElement("div");
            container.setAttribute(
                "data-cslp",
                "all_fields.bltapikey.en-us.single_line_textbox_multiple_"
            );
            container.getBoundingClientRect = vi
                .fn()
                .mockReturnValue(mockDomRect.singleHorizontal());

            firstSingleLineField = document.createElement("p");
            firstSingleLineField.setAttribute(
                "data-cslp",
                "all_fields.bltapikey.en-us.single_line_textbox_multiple_.0"
            );
            firstSingleLineField.getBoundingClientRect = vi
                .fn()
                .mockReturnValue(mockDomRect.singleLeft());

            secondSingleLineField = document.createElement("p");
            secondSingleLineField.setAttribute(
                "data-cslp",
                "all_fields.bltapikey.en-us.single_line_textbox_multiple_.1"
            );
            secondSingleLineField.getBoundingClientRect = vi
                .fn()
                .mockReturnValue(mockDomRect.singleRight());

            container.appendChild(firstSingleLineField);
            container.appendChild(secondSingleLineField);
            document.body.appendChild(container);

            visualBuilder = new VisualBuilder();
        });

        afterEach(() => {
            visualBuilder.destroy();
        });

        test("should have outline and custom cursor", async () => {
            await act(() => {
                container.dispatchEvent(mousemoveEvent);
            });
            container.dispatchEvent(mousemoveEvent);
            await waitForHoverOutline();
            expect(container).not.toHaveAttribute("style");

            const hoverOutline = document.querySelector(
                "[data-testid='visual-builder__hover-outline']"
            );
            expect(hoverOutline).toHaveStyle(
                "top: 34px; left: 34px; width: 828px; height: 54.3984375px;"
            );

            const customCursor = document.querySelector(
                `[data-testid="visual-builder__cursor"]`
            );

            expect(customCursor).toHaveAttribute("data-icon", "singleline");
            expect(customCursor?.classList.contains("visible")).toBeTruthy();
        });

        test("should have outline and custom cursor on individual instances", async () => {
            await act(() => {
                firstSingleLineField.dispatchEvent(mousemoveEvent);
            });
            await waitForHoverOutline();
            expect(firstSingleLineField).not.toHaveAttribute("style");
            const hoverOutline = document.querySelector(
                "[data-testid='visual-builder__hover-outline']"
            );
            expect(hoverOutline).toHaveStyle(
                "top: 51px; left: 51px; width: 27.7734375px; height: 20.3984375px;"
            );

            const customCursor = document.querySelector(
                `[data-testid="visual-builder__cursor"]`
            );

            expect(customCursor).toHaveAttribute("data-icon", "singleline");
            expect(customCursor?.classList.contains("visible")).toBeTruthy();
        });
    });
});
