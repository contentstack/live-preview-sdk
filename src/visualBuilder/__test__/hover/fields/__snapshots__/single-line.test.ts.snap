// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`When an element is hovered in visual builder mode > single line field (multiple) > should have outline and custom cursor 1`] = `
<div
  data-cslp="all_fields.bltapikey.en-us.single_line_textbox_multiple_"
>
  <p
    data-cslp="all_fields.bltapikey.en-us.single_line_textbox_multiple_.0"
  />
  <p
    data-cslp="all_fields.bltapikey.en-us.single_line_textbox_multiple_.1"
  />
</div>
`;

exports[`When an element is hovered in visual builder mode > single line field (multiple) > should have outline and custom cursor 2`] = `
<div
  class="visual-builder__hover-outline visual-builder__hover-outline--unclickable go217287799 go1719569523"
  data-testid="visual-builder__hover-outline"
  style="top: 34px; left: 34px; width: 828px; height: 54.3984375px;"
/>
`;

exports[`When an element is hovered in visual builder mode > single line field (multiple) > should have outline and custom cursor 3`] = `
<div
  class="go1688149084 visual-builder__cursor visible"
  data-icon="singleline"
  data-testid="visual-builder__cursor"
>
  <div
    class="visual-builder__cursor-wrapper"
  >
    <div
      class="visual-builder__cursor-pointer go3046085498"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0.0167948 0.602214C-0.0784404 0.246792 0.246794 -0.0784402 0.602216 0.0167948L15.6457 4.04767C16.0914 4.1671 16.1259 4.78625 15.6962 4.95447L8.17161 7.90048C8.04753 7.94906 7.94936 8.04723 7.90078 8.17131L4.95449 15.6962C4.78627 16.1259 4.16712 16.0914 4.04769 15.6457L0.0167948 0.602214Z"
          fill="#5D50BE"
        />
      </svg>
    </div>
    <div
      class="visual-builder__cursor-icon go4261356330"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M5.94958 7.8294C5.85419 7.6199 5.64077 7.48967 5.41084 7.50064C5.1809 7.51162 4.98086 7.66158 4.90585 7.87921L3.19036 12.8566L2.2835 15.3717C2.17813 15.664 2.32962 15.9863 2.62186 16.0917C2.9141 16.197 3.23643 16.0455 3.34181 15.7533L3.95144 14.0625H7.53908L8.29768 15.7888C8.42266 16.0732 8.75454 16.2025 9.03895 16.0775C9.32336 15.9525 9.4526 15.6206 9.32762 15.3362L8.22073 12.8173L8.21769 12.8105L5.94958 7.8294ZM7.03937 12.9375L5.51002 9.57881L4.35242 12.9375H7.03937Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M11.625 8.0625C11.625 7.75184 11.3732 7.5 11.0625 7.5C10.7518 7.5 10.5 7.75184 10.5 8.0625V15.5625C10.5 15.8732 10.7518 16.125 11.0625 16.125C11.3731 16.125 11.6249 15.8733 11.625 15.5627C12.0951 15.9158 12.6793 16.125 13.3125 16.125C14.8658 16.125 16.125 14.8658 16.125 13.3125C16.125 11.7592 14.8658 10.5 13.3125 10.5C12.6793 10.5 12.0951 10.7092 11.625 11.0623V8.0625ZM11.625 13.3116C11.625 13.3119 11.625 13.3122 11.625 13.3125C11.625 13.3128 11.625 13.3131 11.625 13.3134C11.6255 14.245 12.3808 15 13.3125 15C14.2445 15 15 14.2445 15 13.3125C15 12.3805 14.2445 11.625 13.3125 11.625C12.3808 11.625 11.6255 12.38 11.625 13.3116Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M19.9448 10.5025C20.4222 10.4825 20.8969 10.5844 21.324 10.7988C21.6017 10.9381 21.7138 11.2762 21.5744 11.5538C21.4351 11.8315 21.0971 11.9436 20.8194 11.8043C20.5631 11.6757 20.2783 11.6145 19.9919 11.6265C19.7054 11.6385 19.4267 11.7233 19.1821 11.8729C18.9375 12.0224 18.735 12.2319 18.5938 12.4814C18.4526 12.731 18.3773 13.0124 18.375 13.2991C18.3728 13.5858 18.4436 13.8684 18.5808 14.1201C18.718 14.3719 18.9171 14.5845 19.1593 14.738C19.4015 14.8914 19.6788 14.9806 19.9651 14.9972C20.2513 15.0137 20.5371 14.9571 20.7953 14.8326C21.0752 14.6977 21.4114 14.8151 21.5463 15.095C21.6812 15.3748 21.5637 15.711 21.2839 15.846C20.8534 16.0535 20.3772 16.1479 19.9001 16.1203C19.423 16.0927 18.9609 15.944 18.5572 15.6883C18.1535 15.4325 17.8217 15.0781 17.593 14.6585C17.3643 14.2389 17.2463 13.768 17.2501 13.2901C17.2539 12.8123 17.3794 12.3433 17.6147 11.9274C17.8501 11.5115 18.1875 11.1624 18.5952 10.9131C19.0029 10.6638 19.4673 10.5225 19.9448 10.5025Z"
          fill="white"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`When an element is hovered in visual builder mode > single line field (multiple) > should have outline and custom cursor on individual instances 1`] = `
<p
  data-cslp="all_fields.bltapikey.en-us.single_line_textbox_multiple_.0"
/>
`;

exports[`When an element is hovered in visual builder mode > single line field (multiple) > should have outline and custom cursor on individual instances 2`] = `
<div
  class="visual-builder__hover-outline visual-builder__hover-outline--unclickable go217287799 go1719569523"
  data-testid="visual-builder__hover-outline"
  style="top: 51px; left: 51px; width: 27.7734375px; height: 20.3984375px;"
/>
`;

exports[`When an element is hovered in visual builder mode > single line field (multiple) > should have outline and custom cursor on individual instances 3`] = `
<div
  class="go1688149084 visual-builder__cursor visible"
  data-icon="singleline"
  data-testid="visual-builder__cursor"
>
  <div
    class="visual-builder__cursor-wrapper"
  >
    <div
      class="visual-builder__cursor-pointer go3046085498"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0.0167948 0.602214C-0.0784404 0.246792 0.246794 -0.0784402 0.602216 0.0167948L15.6457 4.04767C16.0914 4.1671 16.1259 4.78625 15.6962 4.95447L8.17161 7.90048C8.04753 7.94906 7.94936 8.04723 7.90078 8.17131L4.95449 15.6962C4.78627 16.1259 4.16712 16.0914 4.04769 15.6457L0.0167948 0.602214Z"
          fill="#5D50BE"
        />
      </svg>
    </div>
    <div
      class="visual-builder__cursor-icon go4261356330"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M5.94958 7.8294C5.85419 7.6199 5.64077 7.48967 5.41084 7.50064C5.1809 7.51162 4.98086 7.66158 4.90585 7.87921L3.19036 12.8566L2.2835 15.3717C2.17813 15.664 2.32962 15.9863 2.62186 16.0917C2.9141 16.197 3.23643 16.0455 3.34181 15.7533L3.95144 14.0625H7.53908L8.29768 15.7888C8.42266 16.0732 8.75454 16.2025 9.03895 16.0775C9.32336 15.9525 9.4526 15.6206 9.32762 15.3362L8.22073 12.8173L8.21769 12.8105L5.94958 7.8294ZM7.03937 12.9375L5.51002 9.57881L4.35242 12.9375H7.03937Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M11.625 8.0625C11.625 7.75184 11.3732 7.5 11.0625 7.5C10.7518 7.5 10.5 7.75184 10.5 8.0625V15.5625C10.5 15.8732 10.7518 16.125 11.0625 16.125C11.3731 16.125 11.6249 15.8733 11.625 15.5627C12.0951 15.9158 12.6793 16.125 13.3125 16.125C14.8658 16.125 16.125 14.8658 16.125 13.3125C16.125 11.7592 14.8658 10.5 13.3125 10.5C12.6793 10.5 12.0951 10.7092 11.625 11.0623V8.0625ZM11.625 13.3116C11.625 13.3119 11.625 13.3122 11.625 13.3125C11.625 13.3128 11.625 13.3131 11.625 13.3134C11.6255 14.245 12.3808 15 13.3125 15C14.2445 15 15 14.2445 15 13.3125C15 12.3805 14.2445 11.625 13.3125 11.625C12.3808 11.625 11.6255 12.38 11.625 13.3116Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M19.9448 10.5025C20.4222 10.4825 20.8969 10.5844 21.324 10.7988C21.6017 10.9381 21.7138 11.2762 21.5744 11.5538C21.4351 11.8315 21.0971 11.9436 20.8194 11.8043C20.5631 11.6757 20.2783 11.6145 19.9919 11.6265C19.7054 11.6385 19.4267 11.7233 19.1821 11.8729C18.9375 12.0224 18.735 12.2319 18.5938 12.4814C18.4526 12.731 18.3773 13.0124 18.375 13.2991C18.3728 13.5858 18.4436 13.8684 18.5808 14.1201C18.718 14.3719 18.9171 14.5845 19.1593 14.738C19.4015 14.8914 19.6788 14.9806 19.9651 14.9972C20.2513 15.0137 20.5371 14.9571 20.7953 14.8326C21.0752 14.6977 21.4114 14.8151 21.5463 15.095C21.6812 15.3748 21.5637 15.711 21.2839 15.846C20.8534 16.0535 20.3772 16.1479 19.9001 16.1203C19.423 16.0927 18.9609 15.944 18.5572 15.6883C18.1535 15.4325 17.8217 15.0781 17.593 14.6585C17.3643 14.2389 17.2463 13.768 17.2501 13.2901C17.2539 12.8123 17.3794 12.3433 17.6147 11.9274C17.8501 11.5115 18.1875 11.1624 18.5952 10.9131C19.0029 10.6638 19.4673 10.5225 19.9448 10.5025Z"
          fill="white"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`When an element is hovered in visual builder mode > single line field > should have outline and custom cursor 1`] = `
<p
  data-cslp="all_fields.bltapikey.en-us.single_line"
/>
`;

exports[`When an element is hovered in visual builder mode > single line field > should have outline and custom cursor 2`] = `
<div
  class="go1688149084 visual-builder__cursor visible"
  data-icon="singleline"
  data-testid="visual-builder__cursor"
>
  <div
    class="visual-builder__cursor-wrapper"
  >
    <div
      class="visual-builder__cursor-pointer go3046085498"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0.0167948 0.602214C-0.0784404 0.246792 0.246794 -0.0784402 0.602216 0.0167948L15.6457 4.04767C16.0914 4.1671 16.1259 4.78625 15.6962 4.95447L8.17161 7.90048C8.04753 7.94906 7.94936 8.04723 7.90078 8.17131L4.95449 15.6962C4.78627 16.1259 4.16712 16.0914 4.04769 15.6457L0.0167948 0.602214Z"
          fill="#5D50BE"
        />
      </svg>
    </div>
    <div
      class="visual-builder__cursor-icon go4261356330"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M5.94958 7.8294C5.85419 7.6199 5.64077 7.48967 5.41084 7.50064C5.1809 7.51162 4.98086 7.66158 4.90585 7.87921L3.19036 12.8566L2.2835 15.3717C2.17813 15.664 2.32962 15.9863 2.62186 16.0917C2.9141 16.197 3.23643 16.0455 3.34181 15.7533L3.95144 14.0625H7.53908L8.29768 15.7888C8.42266 16.0732 8.75454 16.2025 9.03895 16.0775C9.32336 15.9525 9.4526 15.6206 9.32762 15.3362L8.22073 12.8173L8.21769 12.8105L5.94958 7.8294ZM7.03937 12.9375L5.51002 9.57881L4.35242 12.9375H7.03937Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M11.625 8.0625C11.625 7.75184 11.3732 7.5 11.0625 7.5C10.7518 7.5 10.5 7.75184 10.5 8.0625V15.5625C10.5 15.8732 10.7518 16.125 11.0625 16.125C11.3731 16.125 11.6249 15.8733 11.625 15.5627C12.0951 15.9158 12.6793 16.125 13.3125 16.125C14.8658 16.125 16.125 14.8658 16.125 13.3125C16.125 11.7592 14.8658 10.5 13.3125 10.5C12.6793 10.5 12.0951 10.7092 11.625 11.0623V8.0625ZM11.625 13.3116C11.625 13.3119 11.625 13.3122 11.625 13.3125C11.625 13.3128 11.625 13.3131 11.625 13.3134C11.6255 14.245 12.3808 15 13.3125 15C14.2445 15 15 14.2445 15 13.3125C15 12.3805 14.2445 11.625 13.3125 11.625C12.3808 11.625 11.6255 12.38 11.625 13.3116Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M19.9448 10.5025C20.4222 10.4825 20.8969 10.5844 21.324 10.7988C21.6017 10.9381 21.7138 11.2762 21.5744 11.5538C21.4351 11.8315 21.0971 11.9436 20.8194 11.8043C20.5631 11.6757 20.2783 11.6145 19.9919 11.6265C19.7054 11.6385 19.4267 11.7233 19.1821 11.8729C18.9375 12.0224 18.735 12.2319 18.5938 12.4814C18.4526 12.731 18.3773 13.0124 18.375 13.2991C18.3728 13.5858 18.4436 13.8684 18.5808 14.1201C18.718 14.3719 18.9171 14.5845 19.1593 14.738C19.4015 14.8914 19.6788 14.9806 19.9651 14.9972C20.2513 15.0137 20.5371 14.9571 20.7953 14.8326C21.0752 14.6977 21.4114 14.8151 21.5463 15.095C21.6812 15.3748 21.5637 15.711 21.2839 15.846C20.8534 16.0535 20.3772 16.1479 19.9001 16.1203C19.423 16.0927 18.9609 15.944 18.5572 15.6883C18.1535 15.4325 17.8217 15.0781 17.593 14.6585C17.3643 14.2389 17.2463 13.768 17.2501 13.2901C17.2539 12.8123 17.3794 12.3433 17.6147 11.9274C17.8501 11.5115 18.1875 11.1624 18.5952 10.9131C19.0029 10.6638 19.4673 10.5225 19.9448 10.5025Z"
          fill="white"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`When an element is hovered in visual builder mode > title field > should have outline and custom cursor 1`] = `
<p
  data-cslp="all_fields.bltapikey.en-us.title"
/>
`;

exports[`When an element is hovered in visual builder mode > title field > should have outline and custom cursor 2`] = `
<div
  class="visual-builder__hover-outline visual-builder__hover-outline--unclickable go217287799 go1719569523"
  data-testid="visual-builder__hover-outline"
  style="top: 51px; left: 51px; width: 27.7734375px; height: 20.3984375px;"
/>
`;

exports[`When an element is hovered in visual builder mode > title field > should have outline and custom cursor 3`] = `
<div
  class="go1688149084 visual-builder__cursor visible"
  data-icon="singleline"
  data-testid="visual-builder__cursor"
>
  <div
    class="visual-builder__cursor-wrapper"
  >
    <div
      class="visual-builder__cursor-pointer go3046085498"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0.0167948 0.602214C-0.0784404 0.246792 0.246794 -0.0784402 0.602216 0.0167948L15.6457 4.04767C16.0914 4.1671 16.1259 4.78625 15.6962 4.95447L8.17161 7.90048C8.04753 7.94906 7.94936 8.04723 7.90078 8.17131L4.95449 15.6962C4.78627 16.1259 4.16712 16.0914 4.04769 15.6457L0.0167948 0.602214Z"
          fill="#5D50BE"
        />
      </svg>
    </div>
    <div
      class="visual-builder__cursor-icon go4261356330"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M5.94958 7.8294C5.85419 7.6199 5.64077 7.48967 5.41084 7.50064C5.1809 7.51162 4.98086 7.66158 4.90585 7.87921L3.19036 12.8566L2.2835 15.3717C2.17813 15.664 2.32962 15.9863 2.62186 16.0917C2.9141 16.197 3.23643 16.0455 3.34181 15.7533L3.95144 14.0625H7.53908L8.29768 15.7888C8.42266 16.0732 8.75454 16.2025 9.03895 16.0775C9.32336 15.9525 9.4526 15.6206 9.32762 15.3362L8.22073 12.8173L8.21769 12.8105L5.94958 7.8294ZM7.03937 12.9375L5.51002 9.57881L4.35242 12.9375H7.03937Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M11.625 8.0625C11.625 7.75184 11.3732 7.5 11.0625 7.5C10.7518 7.5 10.5 7.75184 10.5 8.0625V15.5625C10.5 15.8732 10.7518 16.125 11.0625 16.125C11.3731 16.125 11.6249 15.8733 11.625 15.5627C12.0951 15.9158 12.6793 16.125 13.3125 16.125C14.8658 16.125 16.125 14.8658 16.125 13.3125C16.125 11.7592 14.8658 10.5 13.3125 10.5C12.6793 10.5 12.0951 10.7092 11.625 11.0623V8.0625ZM11.625 13.3116C11.625 13.3119 11.625 13.3122 11.625 13.3125C11.625 13.3128 11.625 13.3131 11.625 13.3134C11.6255 14.245 12.3808 15 13.3125 15C14.2445 15 15 14.2445 15 13.3125C15 12.3805 14.2445 11.625 13.3125 11.625C12.3808 11.625 11.6255 12.38 11.625 13.3116Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M19.9448 10.5025C20.4222 10.4825 20.8969 10.5844 21.324 10.7988C21.6017 10.9381 21.7138 11.2762 21.5744 11.5538C21.4351 11.8315 21.0971 11.9436 20.8194 11.8043C20.5631 11.6757 20.2783 11.6145 19.9919 11.6265C19.7054 11.6385 19.4267 11.7233 19.1821 11.8729C18.9375 12.0224 18.735 12.2319 18.5938 12.4814C18.4526 12.731 18.3773 13.0124 18.375 13.2991C18.3728 13.5858 18.4436 13.8684 18.5808 14.1201C18.718 14.3719 18.9171 14.5845 19.1593 14.738C19.4015 14.8914 19.6788 14.9806 19.9651 14.9972C20.2513 15.0137 20.5371 14.9571 20.7953 14.8326C21.0752 14.6977 21.4114 14.8151 21.5463 15.095C21.6812 15.3748 21.5637 15.711 21.2839 15.846C20.8534 16.0535 20.3772 16.1479 19.9001 16.1203C19.423 16.0927 18.9609 15.944 18.5572 15.6883C18.1535 15.4325 17.8217 15.0781 17.593 14.6585C17.3643 14.2389 17.2463 13.768 17.2501 13.2901C17.2539 12.8123 17.3794 12.3433 17.6147 11.9274C17.8501 11.5115 18.1875 11.1624 18.5952 10.9131C19.0029 10.6638 19.4673 10.5225 19.9448 10.5025Z"
          fill="white"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;
